src/fold/transformations/features.py

```diff
@@ -0,0 +1,43 @@
+from typing import List, Union
+
+import holidays
+import pandas as pd
+
+from ..utils.list import wrap_in_list
+from .base import Transformation, fit_noop
+
+
+class AddHolidayFeatures(Transformation):
+    """
+    Adds holiday features for given regions as a column.
+    """
+
+    properties = Transformation.Properties()
+
+    def __init__(self, country_codes: Union[List[str], str]) -> None:
+        self.country_codes = wrap_in_list(country_codes)
+        self.name = f"AddRegions-{self.country_codes}"
+        self.holidays = [
+            holidays.country_holidays(country_code)
+            for country_code in self.country_codes
+        ]
+
+    def transform(self, X: pd.DataFrame, in_sample: bool) -> pd.DataFrame:
+        return pd.concat(
+            [
+                X.copy(),
```

Comment:
we don't need to copy here! we should try to avoid unnecessary call to .copy(), if possible :)
we only need to copy if we do an in-place operation